HOWTO-new-release.md

```diff
@@ -1,10 +1,18 @@
 How to make a new release
 -------------------------
 
+Proceed very slowly and triple-check everything because everything is
+immutable: never overwrite or rename any version of anything (tags,
+firmware images, directories, tarballs,...) unless you want all users to
+deeply hate you for not knowing which duplicate version they have.  If a
```

Comment:
The first part of the sentence that explain what should not be done is sufficient. We do not need to threaten release developers to align with a guideline :-)

Reply:
Thanks @mwasko . It's more a "law" than a guideline and it has been [violated in the past](https://github.com/thesofproject/sof-bin/issues/25#issuecomment-736778530) so we need a little push but I agree the phrasing was wrong, good catch thanks. Rephrased.